internal/lintcontext/context.go

```diff
@@ -23,22 +23,37 @@ type InvalidObject struct {
 }
 
 // A LintContext represents the context for a lint run.
-type LintContext struct {
+type LintContext interface {
+	GetObjects() []Object
+	GetInvalidObjects() []InvalidObject
+}
+
+type lintContextImpl struct {
 	objects        []Object
 	invalidObjects []InvalidObject
 }
 
 // Objects returns the (valid) objects loaded from this LintContext.
-func (l *LintContext) Objects() []Object {
+func (l *lintContextImpl) GetObjects() []Object {
 	return l.objects
 }
 
+// AddObject adds a valid object to this LintContext
```

Comment:
Nit: function name comments should start with with the function name (matching case and all)